src/Sulu/Component/Content/Document/Subscriber/WorkflowStageSubscriber.php

```diff
@@ -92,7 +92,7 @@ public function setWorkflowStageOnDocument(HydrateEvent $event)
         }
 
         $node = $event->getNode();
-        $locale = $event->getLocale();
+        $locale = $this->documentInspector->getOriginalLocale($document);
```

Comment:
Looks strange for me, to go always on the document original locale in this listener. What value did have the `$event->getLocale()`? and what value did have the `$this->documentInspector->getOriginalLocale($document);` return?
Should it not be a fallback done if the locale is not correct in that case e.g.:

```suggestion
        $locale = $event->getLocale() ?: $this->documentInspector->getOriginalLocale($document);
```

Reply:
if you are loading the locale `en` that is a shadow page with the target `de`, the values look like this:

- `$event->getLocale()`: de
- `$this->documentInspector->getOriginalLocale($document)`: en

the reason for this is the `ShadowLocaleSubscriber`:
https://github.com/sulu/sulu/blob/bb4e0bad01a3bafd7cb9d162e377a159928832e0/src/Sulu/Component/Content/Document/Subscriber/ShadowLocaleSubscriber.php#L128-L131

Reply:
i got the idea to use the `DocumentInspector` from the `ResourceSegmentSubscriber`:
https://github.com/sulu/sulu/blob/bb4e0bad01a3bafd7cb9d162e377a159928832e0/src/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriber.php#L133-L140

should we add a fallback there too?